dask/delayed.py

```diff
@@ -0,0 +1,484 @@
+
+    def __gt__(self, other):
+        return delayed(operator.gt, pure=True)(self, other)
+
+    def __index__(self):
+        return delayed(operator.index, pure=True)(self)
+
+    def __invert__(self, other):
+        return delayed(operator.invert, pure=True)(self, other)
+
+    def __le__(self, other):
+        return delayed(operator.le, pure=True)(self, other)
+
+    def __lshift__(self, other):
+        return delayed(operator.lshift, pure=True)(self, other)
+
+    def __lt__(self, other):
+        return delayed(operator.lt, pure=True)(self, other)
+
+    def __mod__(self, other):
+        return delayed(operator.mod, pure=True)(self, other)
+
+    def __mul__(self, other):
+        return delayed(operator.mul, pure=True)(self, other)
+
+    def __ne__(self, other):
+        return delayed(operator.ne, pure=True)(self, other)
+
+    def __neg__(self):
+        return delayed(operator.neg, pure=True)(self)
+
+    def __or__(self, other):
+        return delayed(operator.or_, pure=True)(self, other)
+
+    def __pos__(self):
+        return delayed(operator.pos, pure=True)(self)
+
+    def __pow__(self, other):
+        return delayed(operator.pow, pure=True)(self, other)
+
+    def __radd__(self, other):
+        return delayed(right(operator.add), pure=True)(self, other)
+
+    def __rand__(self, other):
+        return delayed(right(operator.and_), pure=True)(self, other)
+
+    def __rdiv__(self, other):
+        return delayed(right(operator.floordiv), pure=True)(self, other)
+
+    def __rfloordiv__(self, other):
+        return delayed(right(operator.floordiv), pure=True)(self, other)
+
+    def __rlshift__(self, other):
+        return delayed(right(operator.lshift), pure=True)(self, other)
+
+    def __rmod__(self, other):
+        return delayed(right(operator.mod), pure=True)(self, other)
+
+    def __rmul__(self, other):
+        return delayed(right(operator.mul), pure=True)(self, other)
+
+    def __ror__(self, other):
+        return delayed(right(operator.or_), pure=True)(self, other)
+
+    def __rpow__(self, other):
+        return delayed(right(operator.pow), pure=True)(self, other)
+
+    def __rrshift__(self, other):
+        return delayed(right(operator.rshift), pure=True)(self, other)
+
+    def __rshift__(self, other):
+        return delayed(operator.rshift, pure=True)(self, other)
+
+    def __rsub__(self, other):
+        return delayed(right(operator.sub), pure=True)(self, other)
+
+    def __rtruediv__(self, other):
+        return delayed(right(operator.truediv), pure=True)(self, other)
+
+    def __rxor__(self, other):
+        return delayed(right(operator.xor), pure=True)(self, other)
+
+    def __sub__(self, other):
+        return delayed(operator.sub, pure=True)(self, other)
+
+    def __truediv__(self, other):
+        return delayed(operator.truediv, pure=True)(self, other)
+
+    def __xor__(self, other):
+        return delayed(operator.xor, pure=True)(self, other)
```

Comment:
@jcrist I had to do this because Python 3 treats assigning callable objects to a class differently from assigning callable functions.